subsys/net/lib/sockets/socket_dispatcher.c

```diff
@@ -0,0 +1,419 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+#include <net/socket.h>
+
+#include "sockets_internal.h"
+
+LOG_MODULE_REGISTER(net_sock_dispatcher, CONFIG_NET_SOCKETS_LOG_LEVEL);
+
+/* TODO create actual configs for those */
+#define CONFIG_NET_SOCKET_DISPATCHER_CONTEXT_MAX 4
```

Comment:
At least earlier there was some checks for preprocessor symbols that started with `CONFIG_` as those are reserved for Kconfig only. I suggest using different name in order to avoid confusion to the reader.